tests/server/logs/test_stream.py

```diff
@@ -0,0 +1,579 @@
+
+    async with subscribed(filter) as queue:
+        assert queue is not None
+        assert queue in subscribed.__wrapped__.__globals__["subscribers"]
+        assert subscribed.__wrapped__.__globals__["filters"][queue] == filter
+
+    # Should be cleaned up after context exit
+    assert queue not in subscribed.__wrapped__.__globals__["subscribers"]
+    assert queue not in subscribed.__wrapped__.__globals__["filters"]
+
+
+@pytest.mark.asyncio
+async def test_logs_context_manager():
+    """Test the logs context manager returns an async iterable"""
+    filter = LogFilter()
+
+    async with logs(filter) as log_stream:
+        assert hasattr(log_stream, "__aiter__")
+        # We can't easily test the actual streaming without setting up
+        # the full messaging system, but we can verify the structure
+
+
+@pytest.mark.asyncio
+async def test_logs_consume_timeout():
+    """Test that the logs consumer yields None on timeout"""
+    import asyncio
+    from unittest.mock import patch
+
+    filter = LogFilter()
+
+    async with logs(filter) as log_stream:
+        # Mock the queue.get to raise TimeoutError
+        with patch("asyncio.wait_for", side_effect=asyncio.TimeoutError):
+            async for log in log_stream:
+                assert log is None
+                break  # Exit after first None
+
+
+@pytest.mark.asyncio
+async def test_distributor_message_handler(sample_log1):
+    """Test the distributor message handler"""
+    from unittest.mock import Mock
+
+    from prefect.server.logs.stream import distributor
+
+    # Create a mock message
+    mock_message = Mock()
+    mock_message.data = sample_log1.model_dump_json().encode()
+    mock_message.attributes = {"log_id": "test"}
+
+    # Create a mock subscriber queue
+    import asyncio
+
+    queue = asyncio.Queue()
+    filter = LogFilter()
+
+    # Add to global subscribers
+    from prefect.server.logs.stream import filters, subscribers
+
+    subscribers.add(queue)
+    filters[queue] = filter
+
+    try:
+        async with distributor() as handler:
+            await handler(mock_message)
+
+            # Should have put the log in the queue
+            assert not queue.empty()
+            log = await queue.get()
+            assert log.message == "Test message 1"
+    finally:
+        # Clean up
+        if queue in subscribers:
+            subscribers.remove(queue)
+        if queue in filters:
+            del filters[queue]
+
+
+@pytest.mark.asyncio
+async def test_distributor_message_handler_no_attributes():
+    """Test distributor handles messages without attributes"""
+    from unittest.mock import Mock
+
+    from prefect.server.logs.stream import distributor
+
+    # Create a mock message without attributes
+    mock_message = Mock()
+    mock_message.data = b"test data"
+    mock_message.attributes = None
+
+    async with distributor() as handler:
+        # Should not raise an exception
+        await handler(mock_message)
```

Comment:
A couple problems here:
* We have a `MemoryMessage` that we can use instead of these mocks
* We don't need to be doing this many inline imports in the test suite, just hoist them to the top of the file